status/status.go

```diff
@@ -99,6 +99,11 @@ func FromError(err error) (s *Status, ok bool) {
 	}
 	type grpcstatus interface{ GRPCStatus() *Status }
 	if gs, ok := err.(grpcstatus); ok {
+		if gs == nil {
+			// gs might be nil even after succesful assertion
+			// returning error status
+			return New(codes.Unknown, err.Error()), false
+		}
 		grpcStatus := gs.GRPCStatus()
 		if grpcStatus == nil {
```

Comment:
I would squash this into a short circuit conditional check like this
```suggestion
		if gs == nil || gs.GRPCStatus() == nil {
```